### ISSUE TITLE
Komodo 9 crashed on debug start

### ISSUE THREAD

**Issue by paulsommer** [Komodo/KomodoEdit#209]
I am not able to start a debugging session in Komodo9 for Linux 
When I click "start" or "single step" the debugging pane at the bottom of the windows appears (if not already open) and that's it. Nothing else happens.
When I click start or single step a second time Komodo crashes.

Once in a while it works, but I also have to click twice in any case.

Tested with PHP on Kubuntu 14.4.2 64bit
Version 8.5.x works flawlessly.

**Comment by cgchoffman** [comment]
Next time you reproduce this please make sure to grab your logs before restarting
~/.komodoide/9.0/pystderr.log.  Copy and paste here in a code section or as a gist.

Could you elaborate on how Komodo Crashes?  Hang? Crash and program terminates?  Partial breakage?

Also, does a dialog appear to submit a crash report?  If yes, if you could get us the email address you submitted it with we'll be able to locate your specific crash report.

**Comment by paulsommer** [comment]
Thanks for your response.
The program crashes completely offering me to send a debug report. I did that twice. The second one had my mailaddress attached, but I never got a reply. The address was <my initials>@edv-sommer.de (sorry for the cryptics, but I don't want to feed the spam bots :)

I'll look for pystderr.log the next time I play with V9.0 (right now I'm short of time and the last few tries worked OK except of the need to click twice)

**Comment by Naatan** [comment]
Have you tried 9.0.1? It resolved an issue with the initialization of the debug process.

**Comment by paulsommer** [comment]
@Nataan: Just tried 9.01. It crashes. :(
@cgchoffman: attached is the pystderr.log of that crash

Hmmm... How can I attach a text file here???
Please look here (excuse the certificate warning :) )
https://owncloud.edv-sommer.de/public.php?service=files&t=2338da682b46f042b1bb06f9cff90bba

**Comment by Naatan** [comment]
Please try Komodo 9.1

**Comment by paulsommer** [comment]
9.1 is somewhat better but still not usable for everyday use.

Most of the time the first click to "run" or "single step" only opens the
debugging pane.
The second click to "run" or "single step" then starts/single steps into the
script or crashes Komodo. I cannot find any systematic in this. It's just like
a random generator.

**Comment by paulsommer** [comment]
Naatan, could you please reopen this ticket?

**Comment by Naatan** [comment]
Could you provide test case to reproduce the problem?

**Comment by paulsommer** [comment]
Test case:
Load a PHP or perl file.
Click "single step" to enter debugging mode. (the debug pane opens but the script doesn't start)
Click "single step" again (now the script starts single stepping OR Komodo crashes)

My system is Kubuntu 14.04 64 bit

If I can provide any further information, please let me know.
If it helps I am willing to participate in teamviewer session (or similar) so you can have a look to the living system.

**Comment by Naatan** [comment]
Can you reproduce this by invoking the debugger using Debug > Go?

**Comment by paulsommer** [comment]
Right now I cannot reproduce the crashes at all (tried a couple of times).
Still there is this strange behaviour (all the time) that with the first "Go" nothing happens beside the debug pane at the bottom appears. The script starts only after the second "Go".

I'll give Ver 9.1 another try when I have to code something (not in the moment) and come bach to you when I found something new.

Oh. I see there's a 9.2.1 out

**Comment by cgchoffman** [comment]
> Still there is this strange behaviour (all the time) that with the first "Go" nothing happens beside the debug pane at the bottom appears. The script starts only after the second "Go".

This is a known and fixed bug.  I believe this is fixed in 9.2.1.

**Comment by paulsommer** [comment]
Sorry, it's not. Still the first "Go" brings the debug pane to front. Only the second "Go" shows the "debug options" dialog ans starts the script.
However, Komodo didn't crash on my last tests.

**Comment by cgchoffman** [comment]
try 9.3 then: http://downloads.activestate.com/Komodo/nightly/komodoide/latest/

**Comment by paulsommer** [comment]
Still the same.

**Comment by cgchoffman** [comment]
I'm not able to reproduce that.  The debugger in 9.3 starts up fine for me without additional clicks.

**Comment by paulsommer** [comment]
ActiveState Komodo IDE
« license info
Licensed to Paul Sommer (Commercial license, serial number Sxxxxxxxxxxx).
« build info
Komodo IDE, version 9.3.0, build 88074, platform linux-x86_64. Built on Thu Oct 22 02:24:33 2015.
copy build info

**Comment by cgchoffman** [comment]
Alright, thank you for double checking.  Can you file a different bug for this new issue please?

**Comment by paulsommer** [comment]
So up to now I had no new crashes. I will open a new ticket for the "click twice to start" issue.
Thanks for any help and for the fine product.

**Comment by paulsommer** [comment]
Oh jeh! With the last nightly (from 12/7/15) Komodo crashes again upon debug start. It doesn't crash every time but at about every 2nd to 5th time.
Unfortunately I didn't save the previous version. Is it possible to download this again?

**Comment by Naatan** [comment]
Could you share your error log at the time of the crash? (Before restarting Komodo).

**Comment by Naatan** [comment]
You can find previous versions here:

http://downloads.activestate.com/Komodo/releases/

But note that without more information from you this issue cannot be resolved.

**Comment by dankasak** [comment]
I'm having the same crashing while launching a debugging session on Linux.

Komodo IDE, version 9.3.2, build 88191, platform linux-x86_64. Built on Thu Dec 10 13:00:48 2015.
Licensed to Daniel Kasak (Commercial license, serial number S2D1DAD665DC).

I've been submitting crash logs. Any word on a fix?

**Comment by Defman21** [comment]
Remove your serial number please.

**Comment by paulsommer** [comment]
Oh, good to see I'm not the only one :)
I still have that problem, too (though only every 5th time or so)

**Comment by mitchell-as** [comment]
@dankasak Please open a new ticket with more information (what language, what steps, your Help > Troubleshooting > View Log File, etc.). Crash reports provide zero context and are not very helpful. Thanks!
